### PR TITLE
Library Main Page README Updates

### PR DESCRIPTION
**Pull request by gshvang**
This PR contains changes to the Main page readme of the Library.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,47 @@
-# AWS IoT SigV4 Utility Library
+# AWS IoT SigV4 Library
 
 **Note** This library is currently under development.
 
-The AWS IoT SigV4 Library is a standalone utility for generating a signature and authorization header according to the specifications of the [Signature Version 4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) signing process. This utility is an optional addition to applications sending direct HTTP requests to AWS services requiring SigV4 authentication.
+The AWS IoT SigV4 Library is a standalone library for generating authorization header and signature according to the specifications of the [Signature Version 4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) signing process. Authorization header is the required for authentication when sending HTTP requests to AWS. This Library is an optional addition to applications sending direct HTTP requests to AWS services requiring SigV4 authentication. This library has no dependencies on any additional libraries other than the standard C library. This library is distributed under the MIT Open Source License.
```
Comment (archigup) [inline README.md:5]:
```suggestion
The AWS IoT SigV4 Library is a standalone library for generating authorization headers and signatures according to the specifications of the [Signature Version 4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) signing process. Authorization headers are required for authentication when sending HTTP requests to AWS. This library can optionally be used by applications sending direct HTTP requests to AWS services requiring SigV4 authentication. This library has no dependencies on any additional libraries other than the standard C library. This library is distributed under the MIT Open Source License.
```

File: README.md [hunk 2]
```diff
@@ -1,12 +1,47 @@
...
 
+This library has gone through code quality checks including verification that no function has a GNU Complexity score over 8, and checks against deviations from mandatory rules in the MISRA coding standard. Deviations from the MISRA C:2012 guidelines are documented under MISRA Deviations. This library has also undergone static code analysis using Coverity static analysis, and validation of memory safety through the CBMC automated reasoning tool.
+
+See memory requirements for this library [here][a9].
+
+[a9]: ./docs/doxygen/include/size_table.md
```
Comment (archigup) [inline README.md:11]:
```suggestion
See memory requirements for this library [here][memory_table].

[memory_table]: ./docs/doxygen/include/size_table.md
```

File: README.md [hunk 3]
```diff
@@ -1,12 +1,47 @@
...
+
+## AWS IoT SigV4 Library Config File
+The AWS IoT SigV4 Library exposes build configuration
+macros that are required for building the library. A list of all the
+configurations and their default values are defined in
+[sigv4_config_defaults.h][b1]. To provide custom values for the
+configuration macros, a config file named `sigv4_config.h` can be
+provided by the application to the library.
+
+[b1]: source/include/sigv4_config_defaults.h
```
Comment (archigup) [inline README.md:21]:
```suggestion
## AWS IoT SigV4 Library Config File
The AWS IoT SigV4 library exposes build configuration
macros that are required for building the library. A list of all the
configurations and their default values are defined in
[sigv4_config_defaults.h][default_config]. To provide custom values for the
configuration macros, a config file named `sigv4_config.h` can be
provided by the application to the library.

[default_config]: source/include/sigv4_config_defaults.h
```

File: README.md (hunk 1 above)
Comment (aggarw13) [reply README.md:5]: Suggest removing the IoT prefix

File: README.md [hunk 4]
```diff
@@ -1,12 +1,47 @@
...
+
+**OR**
+
+* Defining the `SIGV4_DO_NOT_USE_CUSTOM_CONFIG` preprocessor macro
+  for the library build.
+  
 ## Building the SigV4 Library
 
-The [source](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk/tree/main/source) directory contains all of the source files required to build the SigV4 Library. The [source/include](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk/tree/main/source/include) folder should be added to the compiler's include path.
+The sigV4FilePaths.cmake file contains the information of all source files and the header include paths required to build the SigV4 library.
```
Comment (aggarw13) [inline README.md:40]:
```suggestion
The sigV4FilePaths.cmake file contains information of all the source files and header include paths required to build the SigV4 library.
```

**Review by archigup — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (gshvang) [reply README.md]: Shall I remove it now or first discuss it with Tanmoy.

File: README.md [hunk 5]
```diff
@@ -34,7 +69,7 @@ To use CMake, please refer to the [sigV4FilePaths.cmake](https://github.com/aws/
 
 ## Reference examples
 
-The AWS IoT Embedded C-SDK repository contains [demos](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/main/demos/http) showing the use of the AWS IoT SigV4 Client Library on a POSIX platform.
+The AWS IoT Embedded C-SDK repository contains [demos](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/main/demos/http/http_demo_s3_download) showing the use of the AWS IoT SigV4 Client Library on a POSIX platform.
```
Comment (aggarw13) [inline README.md:72]:
```suggestion
The AWS IoT Embedded C-SDK repository contains [HTTP demos](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/main/demos/http/http_demo_s3_download) showing the use of the AWS SigV4 Library on a POSIX platform to authenticate HTTP requests to AWS S3 service.
```

File: README.md [hunk 6]
```diff
@@ -1,12 +1,47 @@
-# AWS IoT SigV4 Utility Library
+# AWS IoT SigV4 Library
```
Comment (aggarw13) [inline README.md:1]:
```suggestion
# AWS SigV4 Library
```

File: README.md [hunk 7]
```diff
@@ -1,12 +1,47 @@
...
 
 **Note** This library is currently under development.
 
-The AWS IoT SigV4 Library is a standalone utility for generating a signature and authorization header according to the specifications of the [Signature Version 4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) signing process. This utility is an optional addition to applications sending direct HTTP requests to AWS services requiring SigV4 authentication.
+The AWS IoT SigV4 Library is a standalone library for generating authorization headers and signatures according to the specifications of the [Signature Version 4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) signing process. Authorization headers are required for authentication when sending HTTP requests to AWS. This library can optionally be used by applications sending direct HTTP requests to AWS services requiring SigV4 authentication. This library has no dependencies on any additional libraries other than the standard C library. This library is distributed under the MIT Open Source License.
```
Comment (aggarw13) [inline README.md:5]:
```suggestion
The AWS SigV4 Library is a standalone library for generating authorization headers and signatures according to the specifications of the [Signature Version 4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) signing process. Authorization headers are required for authentication when sending HTTP requests to AWS. This library can optionally be used by applications sending direct HTTP requests to AWS services requiring SigV4 authentication. This library has no dependencies on any additional libraries other than the standard C library. This library is distributed under the MIT Open Source License.
```

**Review by archigup — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -1,12 +1,47 @@
-# AWS IoT SigV4 Utility Library
+# AWS IoT SigV4 Library
 
 **Note** This library is currently under development.
 
-The AWS IoT SigV4 Library is a standalone utility for generating a signature and authorization header according to the specifications of the [Signature Version 4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) signing process. This utility is an optional addition to applications sending direct HTTP requests to AWS services requiring SigV4 authentication.
+The AWS SigV4 Library is a standalone library for generating authorization headers and signatures according to the specifications of the [Signature Version 4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) signing process. Authorization headers are required for authentication when sending HTTP requests to AWS. This library can optionally be used by applications sending direct HTTP requests to AWS services requiring SigV4 authentication. This library has no dependencies on any additional libraries other than the standard C library. This library is distributed under the MIT Open Source License.
```
Comment (tanmoysen) [inline README.md:5]: ...specifications of the AWS Signature Version 4.....

File: README.md [hunk 9]
```diff
@@ -1,12 +1,47 @@
...
 
+This library has gone through code quality checks including verification that no function has a GNU Complexity score over 8, and checks against deviations from mandatory rules in the MISRA coding standard. Deviations from the MISRA C:2012 guidelines are documented under MISRA Deviations. This library has also undergone static code analysis using Coverity static analysis, and validation of memory safety through the CBMC automated reasoning tool.
+
+See memory requirements for this library [here][memory_table].
+
+[memory_table]: ./docs/doxygen/include/size_table.md
+
+## AWS IoT SigV4 Library Config File
```
Comment (tanmoysen) [inline README.md:13]: Let's use AWS SignV4 throughout

File: README.md [hunk 10]
```diff
@@ -34,7 +69,7 @@ To use CMake, please refer to the [sigV4FilePaths.cmake](https://github.com/aws/
 
 ## Reference examples
 
-The AWS IoT Embedded C-SDK repository contains [demos](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/main/demos/http) showing the use of the AWS IoT SigV4 Client Library on a POSIX platform.
+The AWS IoT Embedded C-SDK repository contains [HTTP demos](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/main/demos/http) showing the use of the AWS SigV4 Library on a POSIX platform to authenticate HTTP requests to AWS S3 service.
```
Comment (tanmoysen) [inline README.md:72]: The AWS IoT Device SDK for Embedded C (C-SDK) repository contains....

**Pull request merged**